service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -85,12 +88,35 @@ public final HonoClient getRegistrationServiceClient() {
         return registration;
     }
 
+    /**
+     * Sets the client to use for connecting to the Credentials API (which may be offered by the Device Registry component).
+     *
+     * @param credentialsServiceClient The client.
+     * @throws NullPointerException if the client is {@code null}.
+     */
+    @Qualifier(CredentialsConstants.CREDENTIALS_ENDPOINT)
+    @Autowired
```

Comment:
IMHO it will not be uncommon to have the Credentials API being implemented together with the Device Registration API. Hence, it makes sense to allow for using the same client (`deviceRegistry`) to be used for both.
This also needs to be reflected in the `connectToCredentialsService()` method below ...